cub/detail/exec_check_disable.cuh

```diff
@@ -0,0 +1,41 @@
+/*
+*  Copyright 2021 NVIDIA Corporation
+*
+*  Licensed under the Apache License, Version 2.0 (the "License");
+*  you may not use this file except in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+*/
+
+#pragma once
+
+#include <cub/util_compiler.cuh>
+
+/**
+ * \def CUB_EXEC_CHECK_DISABLE
+ * Wrapper around `#pragma nv_exec_check_disable`.
+ */
+
+// #pragma nv_exec_check_disable is only recognized by NVCC.
+#if defined(__CUDACC__) && \
```

Comment:
Nitpick: I'd like to see something like:

```cuda
#if CUB_DEVICE_COMPILER == CUB_DEVICE_COMPILER_NVCC
/// ... 
#endif 
```

But the `CUB_DEVICE_COMPILER` macro seems to have a different meaning. It uses NVCC as a device compiler even in the case of clang usage. I haven't found any usages of `CUB_DEVICE_COMPILER`. If you have some time to repurpose it to actually define which compiler is used for device code, it would be possible to rewrite this macro as shown above. 

If this is a hotfix, feel free to ignore this suggestion and we'll clean this up later.

Reply:
Following up on our conversation, we'll clean this up in a future PR.